src/mlj_flux.jl

```diff
@@ -0,0 +1,344 @@
+
+    # initiate history:
+    n_batches = length(y)
+
+    parameters = Flux.params(chain)
+
+    # initial loss:
+    losses = (
+        loss(chain(X[i]), y[i]) + penalty(parameters) / n_batches for i in 1:n_batches
+    )
+    history = [mean(losses)]
+
+    for i in 1:epochs
+        current_loss = MLJFlux.train!(
+            model::MLJFlux.MLJFluxModel, penalty, chain, optimiser, X, y
+        )
+        verbosity < 2 || @info "Loss is $(round(current_loss; sigdigits=4))"
+        verbosity != 1 || next!(meter)
+        push!(history, current_loss)
+    end
+
+    la = model.la
+
+    # fit the Laplace model:
+    fit!(la, zip(X, y); model.fit_params...)
+    optimize_prior!(la; verbose=false, n_steps=100)
+
+    model.la = la
+
+    return chain, history
+end
+
+function MMI.clean!(model::LaplaceApproximation)
+    warning = ""
+    if model.lambda < 0
+        warning *= "Need `lambda ≥ 0`. Resetting `lambda = 0`. "
+        model.lambda = 0
+    end
+    if model.alpha < 0 || model.alpha > 1
+        warning *= "Need alpha in the interval `[0, 1]`. " * "Resetting `alpha = 0`. "
+        model.alpha = 0
+    end
+    if model.epochs < 0
+        warning *= "Need `epochs ≥ 0`. Resetting `epochs = 10`. "
+        model.epochs = 10
+    end
+    if model.batch_size < 0
+        warning *= "Need `batch_size ≥ 0`. Resetting `batch_size = 1`. "
+        model.batch_size = 1
+    end
+    if model.acceleration isa CUDALibs && gpu_isdead()
+        warning *=
+            "`acceleration isa CUDALibs` " * "but no CUDA device (GPU) currently live. "
+    end
+    if !(model.acceleration isa CUDALibs || model.acceleration isa CPU1)
+        warning *= "`Undefined acceleration, falling back to CPU`"
+        model.acceleration = CPU1()
+    end
+    if model.likelihood ∉ (:classification, :regression)
+        warning *=
+            "Need `likelihood ∈ (:classification, :regression)`. " *
+            "Resetting `likelihood = :classification`. "
+        model.likelihood = :classification
+    end
+    if model.subset_of_weights ∉ (:all, :last_layer, :subnetwork)
+        warning *=
+            "Need `subset_of_weights ∈ (:all, :last_layer, :subnetwork)`. " *
+            "Resetting `subset_of_weights = :all`. "
+        model.subset_of_weights = :all
+    end
+    if model.hessian_structure ∉ (:full, :diagonal)
+        warning *=
+            "Need `hessian_structure ∈ (:full, :diagonal)`. " *
+            "Resetting `hessian_structure = :full`. "
+        model.hessian_structure = :full
+    end
+    if model.backend ∉ (:GGN, :EmpiricalFisher)
+        warning *=
+            "Need `backend ∈ (:GGN, :EmpiricalFisher)`. " * "Resetting `backend = :GGN`. "
+        model.backend = :GGN
+    end
+    if model.link_approx ∉ (:probit, :plugin)
+        warning *=
+            "Need `link_approx ∈ (:probit, :plugin)`. " *
+            "Resetting `link_approx = :probit`. "
+        model.link_approx = :probit
+    end
```

Comment:
Is this standard in MLJ to reset to defaults and give a warning instead of throwing an error?

Reply:
Indeed it is, this is the intended use of the method, it restores the model to a usable state, if it has been illegally modified, without throwing any errors.